pkg/core/resources/apis/mesh/proxytemplate_validator.go

```diff
@@ -0,0 +1,80 @@
+package mesh
+
+import (
+	"fmt"
+	"github.com/Kong/kuma/api/mesh/v1alpha1"
+	"github.com/Kong/kuma/pkg/core/validators"
+	"github.com/Kong/kuma/pkg/xds/template"
+	"strings"
+)
+
+var availableProfiles map[string]bool
+var availableProfilesMsg string
+
+func init() {
+	profiles := []string{}
+	availableProfiles = map[string]bool{}
+	for _, profile := range template.AvailableProfiles {
+		availableProfiles[profile] = true
+		profiles = append(profiles, profile)
+	}
+	availableProfilesMsg = strings.Join(profiles, ",")
+}
+
+func (t *ProxyTemplateResource) Validate() error {
+	var verr validators.ValidationError
+	verr.AddError("", validateImports(t.Spec.Imports))
```

Comment:
This is a bit awkward. Ideally I could put here `"imports"` and later
```
verr.AddViolationAt(validators.IndexedAt(i), "cannot be empty")
```
but then the dot fill be added `imports.[0]`.

Alternatively I can refactor this to
```
verr.AddError(RootList("imports"), validateImports(t.Spec.Imports))
```
and change every occurance from `"field"` to `RootField("field")`

unless there is better way to do it?

Reply:
`When should I use "" ?` When you want to add errors without root path. I see that you've added `Add` func for this. Switched for this for now.

`Why aren't 2 paths always joined correctly ?` Because of the indexing. If you want to have root path as "imports" `AddError("imports", verr)` and then add errors with only using index (and not `imports`, you'll end up with `imports.[0]`. 

I don't have a good idea how to refactor it for now.